### PR TITLE
roachtest: explore further shutdown options in ldr node shutdown roachtest

### PR DESCRIPTION
**Pull request by msbutler**
This patch introduces more random branching in the LDR shutdown roachtest:
- the test randomly chooses to shut down a node in one cluster during a bidrectional LDR setup.
- half of the time, that node is the coordinator of the LDR job
- half of the time, the shutdown is run with sigkill

Epic: none

Release note: none

### CODE REVIEW

File: pkg/cmd/roachtest/tests/logical_data_replication.go [hunk 1]
```diff
@@ -431,6 +431,48 @@ func TestLDROnNodeShutdown(
...
+	} else {
+		t.L().Printf("Shutting down node on left side")
+		shutdownSide = setup.left
+	}
+	if setup.rng.Intn(2) == 0 {
+		shutdownNode = findNonGatewayNode(shutdownSide, setup.rng)
+		t.L().Printf("Shutting down worker node %d", shutdownNode)
+	} else {
+		// We're assuming the gateway node is coordinating the ingestion job, as it
+		// served the create logical repl stmt.
```
Comment (dt) [inline pkg/cmd/roachtest/tests/logical_data_replication.go:457]: Is this implied? I thought anyone could win the race to adopt after a `CreateAdoptableJobWithTxn` call?

Comment (msbutler) [reply pkg/cmd/roachtest/tests/logical_data_replication.go]: good call. just added logic to control for this.

**Review by dt — APPROVED** [review]

**Comment by msbutler** [comment]
tftr!

bors r=dt

**Pull request merged**